File: backendCompteurApp/src/main/java/com/compteurapp/backendcompteurapp/model/UserDB.java
package com.compteurapp.backendcompteurapp.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "users")
@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class UserDB {

    @Id
    @Column
    private String id;

    @Column
    private String email;

    @Column
    private String firstname;

    @Column
    private String lastname;

    @Column
    private String username;

    @Column
    private String role;

    @Column
    private String tva;

    @Column
    private String phoneNumber;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToOne(mappedBy = "user", fetch = FetchType.LAZY)
    @JsonIgnore
    private Photo photo;
    
    @OneToOne
    @JoinColumn(name = "adresse_id")
    private Adresse adresse;

}

File: compteurAppFrontend/src/models/userDB.ts
import { Adresse } from "./adresse";

export class UserDB {
  id: number | undefined;
  categoryId: string | undefined;
  email: string | undefined;
  firstName: string | undefined;
  lastName: string | undefined;
  phoneNumber: string | undefined;
  role: string | undefined;
  tva: string | undefined;
  userName: string | undefined;
  adresse: Adresse | undefined;
}
